test/vm/fuzzer/fuzzer.cpp

```diff
@@ -602,7 +878,23 @@ static void run_loop(int argc, char **argv)
     for (auto i = 0u; i < args.runs; ++i) {
         std::cerr << std::format(
             "Fuzzing with seed @ {}: {}\n", msg_rev, args.seed);
-        do_run(i, args);
+
+        auto const &run = prepare_run(args);
+        size_t iteration_index = 0;
+        size_t message_index = 0;
+        try {
+            do_run(i, args, run, iteration_index, message_index);
+        }
+        catch (FuzzerAssertFailure const &ex) {
+            // Disable randomness in compilation for shrinking
+            args.deterministic_compilation = true;
```

Comment:
We don't have a good way to debug this since the moves introduced by the compiler hook are only visible in the contract assembly listing and so the virtual stack state must be inferred from it. Also, if we wanted to control the randomness of the compiler hook, we'd need to reset the state of the random number generator at the beginning of every `emit_instr` call so that removing instructions doesn't cause the hook to produce different results for subsequent instructions.

Like for shrinking entire runs, I suppose most of the fuzzer counter-examples don't depend on the hook so I disable it when shrinking.